From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tim Schumacher <timschumi@gmx.de>
Date: Thu, 5 May 2022 18:50:31 +0200
Subject: [PATCH] Extend short scan sets into the full list

We don't support the (apparently nonstandard) short variant of scan
sets, so extend them into a full list manually.
---
 chardev/char.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/chardev/char.c b/chardev/char.c
index 661ad8176a982c641531cdab8e05c3815504cc2f..c9f701ffc59837a0480bb56c63d5f8ef7ee022e6 100644
--- a/chardev/char.c
+++ b/chardev/char.c
@@ -383,11 +383,11 @@ QemuOpts *qemu_chr_parse_compat(const char *label, const char *filename,
     if (strstart(filename, "vc", &p)) {
         qemu_opt_set(opts, "backend", "vc", &error_abort);
         if (*p == ':') {
-            if (sscanf(p+1, "%7[0-9]x%7[0-9]", width, height) == 2) {
+            if (sscanf(p+1, "%7[0123456789]x%7[0123456789]", width, height) == 2) {
                 /* pixels */
                 qemu_opt_set(opts, "width", width, &error_abort);
                 qemu_opt_set(opts, "height", height, &error_abort);
-            } else if (sscanf(p+1, "%7[0-9]Cx%7[0-9]C", width, height) == 2) {
+            } else if (sscanf(p+1, "%7[0123456789]Cx%7[0123456789]C", width, height) == 2) {
                 /* chars */
                 qemu_opt_set(opts, "cols", width, &error_abort);
                 qemu_opt_set(opts, "rows", height, &error_abort);
